### ISSUE TITLE
Remove DiGraph.seededLinearize

### ISSUE THREAD

**Issue by seldridge** [freechipsproject/firrtl#1247]
tl;dr: My `DiGraph.seededLinearize` method doesn't make any sense and should probably be removed.

I'm filing this as a tracking issue in case anybody tries to use this method.

### Checklist

- [x] Did you specify the current behavior?
- [x] Did you specify the expected behavior?
- [x] Did you provide a code example showing the problem?
- [x] Did you describe your environment?
- [x] Did you specify relevant external information?

### What is the current behavior?

`DiGraph.seededLinearize` was a method I added to try and tease out one possible linearization from the space of all possible linearizations. I did this by adding a "seed" which would be used to see the initial DFS. However, seeding a DFS doesn't matter as nodes are being walked **in the order in which the edges are stored**. So, this method doesn't really have a point except it works for all test cases currently explored in the dependency API.

### What is the expected behavior?

The expected behavior is that this would result in a specific order in which nodes are walked. This can be done inside the `seededLinearize` method or with an implicit ordering. However, it likely makes more sense to just remove this method.

If a user that wants a specific ordering can instead, produce a new `DiGraph` with everything in the order they want, e.g., [the approach taken in the Dependency API PR](https://github.com/freechipsproject/firrtl/pull/1123/commits/da7c4293e25281b0eaf837c64f4467f956b52d10#diff-4aa6423289d5fcb0bcd5aee390090c4bR206).

### Steps to Reproduce

(Intentionally poorly documented steps to reproduce because I'm not intending somebody to jump in and fix this. I'm only expecting for this method to be removed in 1.3.0.)

Running the following test case with the old logic will fail: 
 
 - https://github.com/freechipsproject/firrtl/pull/1123/commits/9b9ef2826e0a6b86835620333c93f39dd7c8f966

### Your environment

- `dependency-api-2` branch: 9b9ef2826e0a6b86835620333c93f39dd7c8f966 (this commit will probably disappear due to a rebase)

### External Information

None.
